### PR TITLE
bottles: 51.13 -> 51.15; add Gliczy as maintainer

### PR DESCRIPTION
**Pull request by Gliczy**
https://github.com/bottlesdevs/Bottles/compare/51.13...51.14
https://github.com/bottlesdevs/Bottles/releases/tag/51.14

https://github.com/bottlesdevs/Bottles/compare/51.14...51.15
https://github.com/bottlesdevs/Bottles/releases/tag/51.15

I took the opportunity to add additional changes:
- Added myself as maintainer
- Use `nix-update-script` instead of `gitUpdater` which was downgrading the package
- Move to pkgs/by-name
- Removed `with libs;` and instead use `lib.`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Gliczy** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353501`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
  </ul>
</details>

File: pkgs/by-name/bo/bottles-unwrapped/package.nix [hunk 1]
```diff
@@ -25,17 +24,18 @@
...
 
 python3Packages.buildPythonApplication rec {
   pname = "bottles-unwrapped";
-  version = "51.13";
+  version = "51.15";
 
   src = fetchFromGitHub {
     owner = "bottlesdevs";
     repo = "bottles";
     rev = version;
```
Comment (NotAShelf) [inline pkgs/by-name/bo/bottles-unwrapped/package.nix:37]: You can also switch this to `refs/tags/${version}` while you're at it. Not a blocker, but would be nice to get it out of the way.

Comment (Gliczy) [reply pkgs/by-name/bo/bottles-unwrapped/package.nix]: Took care of that.
Thanks for reviewing this.

**Comment by francocalvo** [comment]
I'll take this as a chance to ask if something changed in Bottles? I used to install games no problem with it, and now I get a write error:

`02b4:err:virtual:allocate_virtual_memory out of memory for allocation, base (nil) size 70d08000
02b4:err:virtual:try_map_free_area mmap() error Cannot allocate memory, range 0x90000000-0xd04000, unix_prot 0.
` 

Looking around, I found people had the same issue on Nix: https://www.reddit.com/r/NixOS/comments/1f3xi3i/wine_has_problems_writing_to_disk/

**Review by NotAShelf — APPROVED** [review]

**Comment by NotAShelf** [comment]
Can't post the nixpkgs-review as it runs out of memory, but builds fine with `nix build`.

**Comment by Gliczy** [comment]
@francocalvo I’ve done some research and found that others have reported somewhat similar issues on different distros and in `wine` itself, not `bottles` specifically. The Reddit post you shared also refers to `wine`, not `bottles`.

Since this PR is unrelated, consider filing an issue (in nixpkgs, unless we can confirm that it's an upstream issue) if you can confirm that this issue only happens with `bottles`.

**Review by keenanweaver — APPROVED** [review]
LGTM. Only difference from the Flatpak version is this core tab, which doesn't do anything. Don't think it hurts anything being there, just strange.

image

**Comment by Gliczy** [comment]
The "Core" tab is already present in the current version, as it is only hidden if `bottles` is running as a flatpak, see: https://github.com/bottlesdevs/Bottles/blob/e93cd189e966ed0daf515f8398f12e2cb6dcf620/bottles/frontend/views/preferences.py#L84.

Taking a look at https://github.com/bottlesdevs/Bottles/blob/e93cd189e966ed0daf515f8398f12e2cb6dcf620/bottles/frontend/ui/preferences.blp#L293, it is supposed to list the runtimes and winebridge. I may look more into it in the future.

**Pull request merged**